### PR TITLE
Send email snapshot

### PR DESCRIPTION
**Pull request by belucid**
https://trello.com/c/fQZUEw0C

This implements the email sharing option for posting new snapshots (stakeholder-updates).

Requires infrastructure PR: https://github.com/belucid/open-company-infrastructure/pull/20

Requires Email Service repo: https://github.com/open-company/open-company-email

**Code Review**

Review the code here in this PR.

Review the code in the infrastructure PR.

Review the code in `master` branch of the email service (it's the initiation of that repo, so no PR there). You're doing the initial code review.

**Testing**

- [x] To test the API:

Ensure you have updated AWS SQS queue settings, e.g.:

```
export AWS_SQS_BOT_QUEUE='https://sqs.us-east-1.amazonaws.com/892554801312/oc-bot-dev-sean'
export AWS_SQS_EMAIL_QUEUE='https://sqs.us-east-1.amazonaws.com/892554801312/oc-email-dev-sean'
```

Then POST a new snapshot to Buffer requesting email distribution to yourself (change the email in this request):

```console
curl -i --header "Authorization: Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VyLWlkIjoic2xhY2s6VTA2U0JUWEpSIiwibmFtZSI6IlNlYW4gSm9obnNvbiIsInJlYWwtbmFtZSI6IlNlYW4gSm9obnNvbiIsImF2YXRhciI6Imh0dHBzOlwvXC9zZWN1cmUuZ3JhdmF0YXIuY29tXC9hdmF0YXJcL2Y1YjhmYzFhZmZhMjY2YzgwNzIwNjhmODExZjYzZTA0LmpwZz9zPTE5MiZkPWh0dHBzJTNBJTJGJTJGc2xhY2suZ2xvYmFsLnNzbC5mYXN0bHkubmV0JTJGN2ZhOSUyRmltZyUyRmF2YXRhcnMlMkZhdmFfMDAyMC0xOTIucG5nIiwiZW1haWwiOiJzZWFuQG9wZW5jb21wYW55LmNvbSIsIm93bmVyIjpmYWxzZSwiYWRtaW4iOnRydWUsIm9yZy1pZCI6InNsYWNrOlQwNlNCTUg2MCJ9.9Q8GNBojQ_xXT0lMtKve4fb5Pdh260oc2aUc-wP8dus" -X POST -d '{"to": "you@you.com", "subject": "hi", "note":"Cool.", "email": true}' --header "Content-Type: application/json" http://localhost:3000/companies/buffer/updates
```

Then checks AWS SQS console, is there a message waiting for your email service in the email SQS queue? Great.

Now try POSTing again, this time with insufficient params:

```console
curl -i --header "Authorization: Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VyLWlkIjoic2xhY2s6VTA2U0JUWEpSIiwibmFtZSI6IlNlYW4gSm9obnNvbiIsInJlYWwtbmFtZSI6IlNlYW4gSm9obnNvbiIsImF2YXRhciI6Imh0dHBzOlwvXC9zZWN1cmUuZ3JhdmF0YXIuY29tXC9hdmF0YXJcL2Y1YjhmYzFhZmZhMjY2YzgwNzIwNjhmODExZjYzZTA0LmpwZz9zPTE5MiZkPWh0dHBzJTNBJTJGJTJGc2xhY2suZ2xvYmFsLnNzbC5mYXN0bHkubmV0JTJGN2ZhOSUyRmltZyUyRmF2YXRhcnMlMkZhdmFfMDAyMC0xOTIucG5nIiwiZW1haWwiOiJzZWFuQG9wZW5jb21wYW55LmNvbSIsIm93bmVyIjpmYWxzZSwiYWRtaW4iOnRydWUsIm9yZy1pZCI6InNsYWNrOlQwNlNCTUg2MCJ9.9Q8GNBojQ_xXT0lMtKve4fb5Pdh260oc2aUc-wP8dus" -X POST -d '{"to": "you@you.com", "email": true}' --header "Content-Type: application/json" http://localhost:3000/companies/buffer/updates
```

Failure response from the API? Checks AWS SQS console, there's no additional message waiting for your email service in the email SQS queue? Great.

- [ ] To test the Email Service:

Now run your email service (see the README in the email service repo and follow the setup/startup steps). Just do what the README says. You are testing the README. Was there sufficient information to get going? Did you get an email. Groovy!

Now try it from the Web UI, sharing a snapshot and selecting email as the distribution. Did you get an email again? Hot damn!

Deploy this branch, the email service, and the web client to staging using the infrastructure PR referenced above. Repeat the UI test. Did you get an email? Hot damn.

- [ ] Merge (API and Infrastructure)
- [ ] Merge web mainline to master
- [ ] Deploy to beta (web, email and API)
- [ ] Test an email snapshot to yourself with Buffer, all good?

### CODE REVIEW

File: src/open_company/lib/email.clj [hunk 1]
```diff
@@ -0,0 +1,38 @@
+(ns open-company.lib.email
+  (:require [amazonica.aws.sqs :as sqs]
+            [taoensso.timbre :as timbre]
+            [schema.core :as schema]
+            [open-company.config :as c]))
+
+(def EmailTrigger
+  {:to schema/Str
```
Comment (martinklepsch) [inline src/open_company/lib/email.clj]: Just as a note: don't forget changing this into a vector.

Comment (belucid) [inline src/open_company/lib/email.clj]: Thanks @martinklepsch I'd forgotten!

**Comment by bago2k4** [comment]
reviewing

**Comment by bago2k4** [comment]
this looks and works good too. merging too 👍

**Pull request merged**